Investigate why JMH tests aren't running

may be related to a PicoCLI update that went in.

It would be nice to have JMH tests running again regularly so that we can get reports out of them.

raised from 2020-12-08 team retro

When running via `gradle` or JAR I got the following results: 
```
# Run progress: 40,00% complete, ETA 00:00:02
# Fork: 3 of 5
# Warmup Iteration   1: WARNING: sun.reflect.Reflection.getCallerClass is not supported. This will impact performance.
<failure>

java.lang.ExceptionInInitializerError
        at tech.pegasys.teku.bls.BLSSecretKey.fromBytes(BLSSecretKey.java:38)
        at tech.pegasys.teku.bls.BLSSecretKey.fromBytesModR(BLSSecretKey.java:53)
        at tech.pegasys.teku.bls.BLSKeyPair.random(BLSKeyPair.java:45)
        at tech.pegasys.teku.bls.BLSPublicKey.random(BLSPublicKey.java:41)
        at tech.pegasys.teku.datastructures.util.DataStructureUtil.lambda$new$0(DataStructureUtil.java:82)
        at tech.pegasys.teku.datastructures.util.DataStructureUtil.randomPublicKeyBytes(DataStructureUtil.java:190)
        at tech.pegasys.teku.datastructures.util.DataStructureUtil.randomValidator(DataStructureUtil.java:576)
        at tech.pegasys.teku.datastructures.util.DataStructureUtil.lambda$randomSSZList$1(DataStructureUtil.java:144)
        at java.base/java.util.stream.Streams$RangeLongSpliterator.forEachRemaining(Streams.java:228)
        at java.base/java.util.stream.LongPipeline$Head.forEach(LongPipeline.java:577)
        at tech.pegasys.teku.datastructures.util.DataStructureUtil.randomSSZList(DataStructureUtil.java:144)
        at tech.pegasys.teku.datastructures.util.BeaconStateBuilder.initDefaults(BeaconStateBuilder.java:130)
        at tech.pegasys.teku.datastructures.util.BeaconStateBuilder.<init>(BeaconStateBuilder.java:69)
        at tech.pegasys.teku.datastructures.util.BeaconStateBuilder.create(BeaconStateBuilder.java:76)
        at tech.pegasys.teku.datastructures.util.DataStructureUtil.randomBeaconState(DataStructureUtil.java:607)
        at tech.pegasys.teku.datastructures.util.DataStructureUtil.randomBeaconState(DataStructureUtil.java:599)
        at tech.pegasys.teku.benchmarks.SSZBenchmark.<clinit>(SSZBenchmark.java:27)
        at tech.pegasys.teku.benchmarks.jmh_generated.SSZBenchmark_BeaconStateSerialization_jmhTest._jmh_tryInit_f_sszbenchmark0_0(SSZBenchmark_BeaconStateSerialization_jmhTest.java:333)
        at tech.pegasys.teku.benchmarks.jmh_generated.SSZBenchmark_BeaconStateSerialization_jmhTest.BeaconStateSerialization_Throughput(SSZBenchmark_BeaconStateSerialization_jmhTest.java:71)
        at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
        at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
        at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
        at java.base/java.lang.reflect.Method.invoke(Method.java:566)
        at org.openjdk.jmh.runner.BenchmarkHandler$BenchmarkTask.call(BenchmarkHandler.java:453)
        at org.openjdk.jmh.runner.BenchmarkHandler$BenchmarkTask.call(BenchmarkHandler.java:437)
        at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
        at java.base/java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:515)
        at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
        at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
        at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
        at java.base/java.lang.Thread.run(Thread.java:834)
Caused by: java.lang.UnsupportedOperationException: No class provided, and an appropriate one cannot be found.
        at org.apache.logging.log4j.LogManager.callerClass(LogManager.java:576)
        at org.apache.logging.log4j.LogManager.getLogger(LogManager.java:601)
        at org.apache.logging.log4j.LogManager.getLogger(LogManager.java:588)
        at tech.pegasys.teku.bls.BLS.<clinit>(BLS.java:46)
        ... 31 more
```
This could be locally tweaked by 
```patch
Index: bls/src/main/java/tech/pegasys/teku/bls/BLS.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- bls/src/main/java/tech/pegasys/teku/bls/BLS.java	(revision b5b3c2ef6f5fe0c7e924e99c2cc1826c6943a3f5)
+++ bls/src/main/java/tech/pegasys/teku/bls/BLS.java	(date 1608645980381)
@@ -43,7 +43,7 @@
  */
 public class BLS {
 
-  private static final Logger LOG = LogManager.getLogger();
+  private static final Logger LOG = LogManager.getLogger(BLS.class);
 
   private static BLS12381 BLS_IMPL;
 ```
 
However if any classes with the same logger initialization are created the problem is getting back 
Note the warning: 
```
# Warmup Iteration   1: WARNING: sun.reflect.Reflection.getCallerClass is not supported.
```

--- a/eth-benchmark-tests/src/jmh/java/tech/pegasys/teku/benchmarks/ShuffleBenchmark.java
+++ b/eth-benchmark-tests/src/jmh/java/tech/pegasys/teku/benchmarks/ShuffleBenchmark.java
@@ -28,7 +28,7 @@ import org.openjdk.jmh.infra.Blackhole;
 import tech.pegasys.teku.datastructures.util.CommitteeUtil;
 import tech.pegasys.teku.util.config.Constants;
 
-@Fork(0)
+@Fork(3)
 @BenchmarkMode(Mode.SingleShotTime)
 @State(Scope.Thread)
 public class ShuffleBenchmark {

--- a/ethereum/datastructures/src/main/java/tech/pegasys/teku/datastructures/util/BeaconStateUtil.java
+++ b/ethereum/datastructures/src/main/java/tech/pegasys/teku/datastructures/util/BeaconStateUtil.java
@@ -82,7 +82,7 @@ import tech.pegasys.teku.util.config.Constants;
 
 public class BeaconStateUtil {
 
-  private static final Logger LOG = LogManager.getLogger();
+  private static final Logger LOG = LogManager.getLogger(BeaconStateUtil.class);
 
   /**
    * For debug/test purposes only enables/disables {@link DepositData} BLS signature verification

--- a/gradle/versions.gradle
+++ b/gradle/versions.gradle
@@ -126,7 +126,7 @@ dependencyManagement {
     dependency 'org.mockito:mockito-core:3.3.3'
     dependency 'org.mockito:mockito-junit-jupiter:3.3.3'
 
-    dependencySet(group: "org.openjdk.jmh", version: "1.25.2") {
+    dependencySet(group: "org.openjdk.jmh", version: "1.27") {
       entry 'jmh-core'
       entry 'jmh-generator-annprocess'
     }

